src/directions/main.ts

```diff
@@ -587,6 +639,48 @@ export default class MapLibreGlDirections extends MapLibreGlDirectionsEvented {
     this.draw();
   }
 
+  protected lastRequestMousePosition = {
+    x: 0,
+    y: 0,
+  };
+
+  protected async liveRefresh(e: MapMouseEvent | MapTouchEvent) {
+    if (
+      Math.abs(this.lastRequestMousePosition?.x - this.currentMousePosition?.x) >
+        (this.configuration.dragThreshold >= 0 ? this.configuration.dragThreshold : 0) ||
+      Math.abs(this.lastRequestMousePosition?.y - this.currentMousePosition?.y) >
+        (this.configuration.dragThreshold >= 0 ? this.configuration.dragThreshold : 0)
+    ) {
+      this.refreshOnMoveIsRefreshing = true;
+      this.lastRequestMousePosition = this.currentMousePosition;
+
+      /*
+       * During liveRefresh, we only care about waypoints, because if dragging a hoverpoint
+       * then it has already been converted to a waypoint on dragDown
+       */
+      if (this.waypointBeingDragged) {
+        /*
+         * If a waypoint has been dragged, we fire a "movewaypoint" just like "onDragUp" does.
+         */
+        const waypointEvent = new MapLibreGlDirectionsWaypointEvent("movewaypoint", e, {
+          index: this._waypoints.indexOf(this.waypointBeingDragged),
+          initialCoordinates: this.waypointBeingDraggedInitialCoordinates,
+        });
+        this.fire(waypointEvent);
+        /*
+         * If the routing request has failed for some reason, restore the waypoint's original position.
```

Comment:
Are we restoring to position before dragDown happened? Or to the last while-moving-request's position?

Reply:
To the position before dragDown happened